dockstore-webservice/src/main/java/io/dockstore/webservice/resources/UserResource.java

```diff
@@ -1014,38 +1014,39 @@ public Limits setUserLimits(@ApiParam(hidden = true)  @Parameter(hidden = true,
     @PUT
     @Timed
     @UnitOfWork
-    @RolesAllowed({"admin", "curator"})
+    @RolesAllowed({"admin"})
     @Path("/{userId}/privileges")
     @Consumes("application/json")
-    @Operation(operationId = "setUserPrivileges", description = "Updates the provided userID to admin or curator status, ADMIN or CURATOR only", security = @SecurityRequirement(name = OPENAPI_JWT_SECURITY_DEFINITION_NAME))
+    @Operation(operationId = "setUserPrivileges", description = "Updates the provided userID to admin or curator status, usable by ADMINs only", security = @SecurityRequirement(name = OPENAPI_JWT_SECURITY_DEFINITION_NAME))
     @ApiResponse(responseCode = HttpStatus.SC_OK + "", description = "Successfully updated user to admin or curator status", content = @Content(schema = @Schema(implementation = User.class)))
     @ApiResponse(responseCode = HttpStatus.SC_FORBIDDEN + "", description = HttpStatusMessageConstants.FORBIDDEN)
     @ApiResponse(responseCode = HttpStatus.SC_NOT_FOUND + "", description = USER_NOT_FOUND_DESCRIPTION)
-    @ApiOperation(value = "Updates the provided userID to admin or curator status, ADMIN or CURATOR only", authorizations = { @Authorization(value = JWT_SECURITY_DEFINITION_NAME) }, response = User.class, hidden = true)
+    @ApiOperation(value = "Updates the provided userID to admin or curator status, usable by ADMINs only", authorizations = { @Authorization(value = JWT_SECURITY_DEFINITION_NAME) }, response = User.class, hidden = true)
     public User setUserPrivilege(@Parameter(hidden = true, name = "user")@Auth User authUser,
                                  @Parameter(name = "User ID", required = true) @PathParam("userId") Long userID,
                                  @Parameter(name = "Set privilege for a user", required = true) PrivilegeRequest privilegeRequest) {
-        User user = userDAO.findById(userID);
-        checkUserExists(user);
+        User targetUser = userDAO.findById(userID);
+        checkUserExists(targetUser);
 
         // This ensures that the user cannot modify their own privileges.
-        if (authUser.getId() == user.getId()) {
+        if (authUser.getId() == targetUser.getId()) {
             throw new CustomWebApplicationException("You cannot modify your own privileges", HttpStatus.SC_FORBIDDEN);
         }
 
-        // If the request's admin setting is different than the admin status of the user that is being modified, and the auth user is not an admin: Throw an error.
-        // This ensures that a curator cannot modify the admin status of any user.
-        if (privilegeRequest.isAdmin() != user.getIsAdmin() && !authUser.getIsAdmin()) {
+        // If the authorized user is not an admin, fail gracefully.
+        // This is a backup test for the @RolesAllowed annotation above, to prevent anarchy in case it is somehow disabled.
+        if (!authUser.getIsAdmin()) {
```

Comment:
If we don't trust `@RolesAllowed`, then we should add the backup check everywhere we use the annotation.

But I don't think we need to.

Reply:
The stakes are so high in this particular method that I figured a backup check was worth it.

Reply:
(I trust @RolesAllowed, but people make mistakes.)